### PR TITLE
Search Clear textbox:issue-1147

### PR DESCRIPTION
**Pull request by Shruti9520**
### Proposed changes in this pull request

Implementing ClearText on the search box mentioned in the screenshot of [issue](https://github.com/Cadasta/cadasta-platform/issues/1147).

Here instead of overriding bootstrap, I used a complete procedure of clearing text because overriding bootstrap technique won't work on browsers like firefox.

Files that are changed here- cadasta/core/static/css/main.scss
                                              cadasta/templates/search/search.html
                                              cadasta/templates/core/base.html

### Risks

-

### Follow up actions

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
- [ ] **Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts.

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
- [ ] **If this is a bug fix, are tests for the issue in place**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: cadasta/templates/search/search.html [hunk 1]
```diff
@@ -189,5 +189,21 @@
...
+<script>
+   
+$(document).ready(function() {
+    function tog(v) { 
+        return v?'addClass':'removeClass';
+    } 
+    $(document).on('input', '.clearable', function() {
+      $(this)[tog(this.value)]('icon');
+      }).on('mousemove', '.icon', function(e) {
+        $(this)[tog(this.offsetWidth-18 < e.clientX-this.getBoundingClientRect().left)]('on_icon');
```
Comment (seav) [inline cadasta/templates/search/search.html]: I do not like this code. There are a lot of hard-coded values. And it does not completely work. In Firefox and Chromium on Linux, the mouse cursor does not become a pointer when it is over the left half of the clear icon. And the space between the icon and the right border is clickable.

File: cadasta/core/static/css/main.scss [hunk 2]
```diff
@@ -978,3 +978,17 @@ h1 .label {
   text-transform: uppercase;
   padding-bottom: 2px;
 }
+
+/* Clear search box
+______________________________________________________________*/
+.clearable {
+    background: url('/static/img/cross.png') no-repeat right -400px center;
```
Comment (seav) [inline cadasta/core/static/css/main.scss]: Why not just add the background image **only** if the class is `icon` instead of adding the image and then hiding it here?

File: cadasta/templates/search/search.html (hunk 1 above)
Comment (Shruti9520) [inline cadasta/templates/search/search.html]: @seav Thank you so much for reviewing my PR. :) 
`(this.offsetWidth-18 < e.clientX-this.getBoundingClientRect().left)]('on_icon')` actually, checks the range of cross icon and the issue that mouse cursor does not become a pointer when it is over the left half of the clear icon can be sovled by changing `this.offsetWidth-18 ` to `this.offsetWidth-30`.
Will you please guide me with some better solution :). Thank you very much. :)

**Comment by Shruti9520** [comment]
@seav Will you please review it and help me with my problem? Thank you very much. :)

File: cadasta/templates/core/base.html [hunk 3]
```diff
@@ -13,7 +13,7 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <title>Cadasta{% block title %}{% endblock %}</title>
-    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.4/jquery.min.js"></script>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>
```
Comment (seav) [inline cadasta/templates/core/base.html]: Maybe I wasn't clear enough. You added this line to import jQuery, but jQuery is already being imported in this file. So your addition is not necessary.

File: cadasta/templates/search/search.html [hunk 4]
```diff
@@ -190,20 +190,20 @@
...
+            return v?'addClass':'removeClass';
+        } 
+        $(document).on('input', '.clearable', function() {
+                $(this)[tog(this.value)]('x');
+                  }).on('mousemove', '.x', function(e) {
+                     $(this)[tog(this.offsetWidth-30< e.clientX-this.getBoundingClientRect().left)]('on_x');
+                      }).on('touchstart click', '.on_x', function(ev) {
+                            ev.preventDefault();
+                           $(this).removeClass('x on_x').val('').change();
+                           history.pushState(history.state, null,
```
Comment (seav) [inline cadasta/templates/search/search.html]: Sorry, now that I have tried this out. I don't think we should be manipulating the URL when the search field is cleared. The URL should match the search results being shown. And clearing the search field using the clear button is not really different from clearing it by selecting the whole text and then pressing the delete key, so the URL should not change.

File: cadasta/templates/search/search.html (hunk 1 above)
Comment (seav) [inline cadasta/templates/search/search.html]: While I don't like it, I don't have any better ideas, so I think this should be OK for now.

**Comment by seav** [comment]
The following feedback has not yet been addressed:
> When I go to the search page by entering a search query via the sidebar, the clear icon does not appear.

**Comment by Shruti9520** [comment]
Hi,

All required changes have been addressed except- "Right part between the cross mark and the border is clickable". I have included a "cross image" , so this problem is encountered. I will soon find a way to get rid of it. Thanks :)

Here are the screenshots-
p1

p2

**Comment by seav** [comment]
Hi @Shruti9520, are you still planning to work on this PR? The following feedback are still not yet resolved:

1. When I go to the search page by entering a search query via the sidebar, the clear icon does not appear.
2. Your JavaScript indentation is quite confusing. Can you use 2 spaces for an indent and properly align the closing brace with block construct's opening line? See the [Google JS style guide](https://google.github.io/styleguide/jsguide.html) for more details.

**Comment by Shruti9520** [comment]
@seav When I go to search bar via sidebar the search icon do appear. Please, see the screenshot below.

**Comment by seav** [comment]
No. Try clicking on the search button after typing something in the search field sidebar. You will be redirected to the search page with the bigger search field filled in but there is no clear button.

**Comment by Shruti9520** [comment]
@seav I am sorry to misunderstanding. Okay, I will try to solve this. :)

**Comment by Shruti9520** [comment]
@seav I enter text in the search box of side navbar and click on the search, it redirected me to a page with a bigger search box.
Earlier- Cross icon appears only after changing the given text.
Now- Cross icon appears on hovering search box.
